### PR TITLE
Add Battery Support for FreeBSD

### PR DESCRIPTION
**Pull request by tudurom**
Fix it.

### CODE REVIEW

File: fetch [hunk 1]
```diff
@@ -1389,6 +1389,11 @@ getbattery () {
             fi
         ;;
 
+        "BSD")
+          battery=$(acpiconf -i 0 | awk -F ':\t' '/Remaining capacity/ {print $2}')
+          battery=${battery//%}
```
Comment (dylanaraps) [inline fetch]: Doesn't this line remove all instances of `%`?

File: fetch [hunk 2]
```diff
@@ -1389,6 +1389,11 @@ getbattery () {
             fi
         ;;
 
+        "BSD")
```
Comment (dylanaraps) [inline fetch]: If this only works on `FreeBSD` then you should do it like this:

```sh
case "$os" in
  "BSD")
      case "$distro" in
          "FreeBSD"*)
              # FreeBSD battery stuff
          ;;
      esac
  ;;
esac
```

**Comment by dylanaraps** [comment]
Other than those two comments LGTM.

**Comment by tudurom** [comment]
There's only one occurence of '%'. Btw can't fix it now.

**Comment by dylanaraps** [comment]
Why are you removing the `%` though? Looking at the PR it seems like the output will just be missing the `%` character.

File: fetch (hunk 2 above)
Comment (dylanaraps) [inline fetch]: The reason `OpenBSD` has a seperate line in `case "$(uname)"` part of the script
is that the output of `uname` om OpenBSD is entirely different to all other BSDs for 
some reason

**Comment by tudurom** [comment]
Trust me, there is a % and it needs to be removed. On Linux there's no %.

**Comment by dylanaraps** [comment]
What I'm trying to say is that the output needs to include the `%` sign. It should
be appearing on Linux too, have a look through the battery function and you'll
see these three lines which add the `%` sign to the output:

```sh
battery="${battery}%"
battery="${batteries[0]}%"
battery="${batteries[0]}%"
```

**Comment by tudurom** [comment]
I will do it later. I'm busy now.

**Comment by dylanaraps** [comment]
I can do it myself if you'd like.

**Comment by tudurom** [comment]
Stay chill...

**Comment by tudurom** [comment]
Done. You can see I fixed a huge bug that occurs when using `fetch --stdout battery`.

**Comment by tudurom** [comment]
Tested, you can merge it.

File: fetch [hunk 3]
```diff
@@ -2083,7 +2091,7 @@ stdout () {
             *)
                 "get$func" 2>/dev/null
                 eval output="\$$func"
-                printf "$output  "
+                echo -n "$output  "
```
Comment (dylanaraps) [inline fetch]: Change this back to `printf`

File: fetch [hunk 4]
```diff
@@ -1401,6 +1401,14 @@ getbattery () {
             fi
         ;;
 
+        "BSD")
+          case "$distro" in
+            "FreeBSD"*)
+          battery=$(acpiconf -i 0 | awk -F ':\t' '/Remaining capacity/ {print $2}')
```
Comment (dylanaraps) [inline fetch]: Fix the indentation here.

**Comment by dylanaraps** [comment]
I left some more comment, also squash the commits into a single one.

**Comment by tudurom** [comment]
1. How do I squash them
2. Why printf? It breaks everything when printing the battery info because
the '%' at the end will be interpreted as a format specifier.

**Comment by dylanaraps** [comment]
Change the printf line to this:

```sh
printf "%s" "$output  "
```

**Comment by tudurom** [comment]
But why not echo?

**Comment by dylanaraps** [comment]
Echo isn't portable.

http://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo

**Comment by dylanaraps** [comment]
The easiest way to squash commits is to do this:

```sh
# Change '3' ti the number of commits to squash
git reset --soft HEAD~3 &&
git commit -m "New commit message"
```

**Comment by tudurom** [comment]
Thanks, TIL!

That commit squash is a great hack. Will do in a minute.

**Comment by dylanaraps** [comment]
Just a heads up: You may need to force push the squashed commit to github.

```sh
git push -f origin master
```

**Comment by tudurom** [comment]
I think I nailed it.

**Comment by tudurom** [comment]
The commit got squashed with the "Added Windows support for resolution" but eh.

**Comment by dylanaraps** [comment]
You'll have to fix that as I'll have to force push to master which 
will break everybody's local copy and I don't want to do that.

Also, you still need to fix the indentation.

If you have any issues I can merge this manually.

**Comment by tudurom** [comment]
Fuck this, I will reset my fork to upstream and readd support.

**Comment by tudurom** [comment]
Done.

**Comment by dylanaraps** [comment]
Thanks. :)

**Comment by tudurom** [comment]
Go write about this in the changelog and in the TODO issue. Next stop: OpenBSD!

**Comment by dylanaraps** [comment]
:+1:

**Pull request merged**